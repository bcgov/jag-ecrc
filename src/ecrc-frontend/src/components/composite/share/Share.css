.share {
  display: flex;
  flex-direction: column;
}
