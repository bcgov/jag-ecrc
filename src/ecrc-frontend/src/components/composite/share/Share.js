import React from "react";

import "./Share.css";
import Button from "../../base/button/Button";

export default function Share({
  previousOrg,
  expiration,
  newOrg,
  clickShare,
  boxChecked
}) {
  const shareButton = {
    label: "Share",
    buttonStyle: "btn ecrc_go_btn",
    buttonSize: "btn",
    type: "submit",
    disabled: !boxChecked
  };

  return (
    <div className="share">
      <span>You have a valid previous CRC.</span>
      <span>It was prepared for: {previousOrg}</span>
      <span>It is valid until: {expiration}</span>
      <p>Would you like to share it with {newOrg}?</p>

      <Button button={shareButton} onClick={clickShare} />
    </div>
  );
}
