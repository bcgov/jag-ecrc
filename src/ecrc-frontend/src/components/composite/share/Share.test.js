import React from "react";
import { create } from "react-test-renderer";

import Share from "./Share";

describe("Share Component", () => {
  test("Matches the snapshot of Share", () => {
    const displayShare = create(
      <Share
        previousOrg="Previous Org"
        expiration="2021-01-04"
        newOrg="New Org"
        clickShare={() => jest.fn()}
        boxChecked
      />
    );
    expect(displayShare.toJSON()).toMatchSnapshot();
  });
});
