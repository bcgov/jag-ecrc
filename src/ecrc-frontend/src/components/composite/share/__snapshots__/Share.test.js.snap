// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share Component Matches the snapshot of Share 1`] = `
<div
  className="share"
>
  <div
    className="declareTitle"
  >
    Share Your Existing Criminal Record Check
  </div>
  <div
    className="declareSection"
  >
    <p>
      If you are an employee or volunteer, you can share the results of your criminal record check at no cost.
    </p>
    <p>
      <b>
        Note:
      </b>
       An organization can decide whether or not they will accept a shared criminal record check result and may ask you to consent to a new criminal record check manually or online.
    </p>
  </div>
</div>
`;
